docs/investor.md

```diff
@@ -0,0 +1,251 @@
+
+Value is represented in terms of an `AssetId`, NOT in FIAT/USD value.
+
+#### `assetId: string`
+
+This is the `AssetId` in CAIP-19 format that is the `Asset` used in `balance`.
+
+Typically, this will be the same as either `underlyingAsset.assetId` or `positionAsset.assetId`
+
+#### `balance: BigNumber`
+
+This is the balance of the total value in the `Opportunity`.
+
+For example, if there is an `Opportunity` that has `FOX` as an `UnderlyingAsset`,
+and there are 10,000,000 FOX deposited into the contract, then balance would be `BigNumber(1e25)`, since `FOX` has a precision of 18.
+
+### `metadata?: MetaData`
+
+This property is a place to store protocol specific information that you want to be made available to custom UI components.
+
+The default UI components for single-sided staking don't look at this data.
+
+## Functions
+
+### `prepareWithdrawal: (input: DepositWithdrawArgs) => Promise<TxType>`
+
+This function should determine whether the requested withdrawal is possible.
+
+This function should validate the user's address, position balance, and any other information needed to determine if it's okay to proceed to the SIGNING step.
+
+#### `input: { address: string, amount: BigNumber }`
+
+`address: string` - The user's wallet address that contains a `PositionAsset` balance.
+
+`amount: BigNumber` - The amount of `PositionAsset` that the user wants to withdraw.
+
+`amount` must be an integer value, without asset precision applied. For example, 1 FOX would be `BigNumber(1e18)`
+
+#### Returns `TxType`
+
+The return of this function is generic and up to the implementer. The restriction is that the output of this function MUST be used as the input to `signAndBroadcast`.
+
+### `prepareDeposit: (input: DepositWithdrawArgs) => Promise<TxType>`
+
+This function should determine whether the requested deposit is possible.
+
+This function should validate the user's address, underlying asset balance, and any other information needed to determine if it's okay to proceed to the SIGNING step.
+
+#### `input: { address: string, amount: BigNumber }`
+
+`address: string` - The user's wallet address that contains a `UnderlyingAsset` balance.
+
+`amount: BigNumber` - The amount of `UnderlyingAsset` that the user wants to withdraw.
+
+`amount` must be an integer value, without asset precision applied. For example, 1 FOX would be `BigNumber(1e18)`
+
+#### Returns `TxType`
+
+The return of this function is generic and up to the implementer. The restriction is that the output of this function MUST be used as the input to `signAndBroadcast`.
+
+### `signAndBroadcast: (input: { wallet: HDWallet, tx: TxType, feePriority?: FeePriority }) => Promise<string>`
+
+This function is responsible for using `wallet` to sign the prepared transaction and broadcast it.
+
+It is recommended that you use a compatible `ChainAdapter` from `@shapeshiftoss/chain-adapters` to simplify the implementation.
```

Comment:
ChainAdapter is implemented on the Investor level. It's a bit confusing having it in the signAndBroadcast docs.

Reply:
> ChainAdapter is implemented on the Investor level. It's a bit confusing having it in the signAndBroadcast docs.

We chose to do it that way but there's nothing in the generic interface that requires that. In fact, the only place we use ChainAdapter is in the `signAndBroadcast`